eZ/Publish/Core/MVC/Symfony/Matcher/DynamicallyConfiguredMatcherFactoryDecorator.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/**
+ * File containing the AbstractMatcherFactory class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\MVC\Symfony\Matcher;
+
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use eZ\Publish\Core\MVC\Symfony\View\View;
+
+/**
+ * Injects dynamic configuration before every matching operation.
+ */
+class DynamicallyConfiguredMatcherFactoryDecorator implements MatcherFactoryInterface
+{
+    /** @var \eZ\Publish\Core\MVC\Symfony\Matcher\MatcherFactoryInterface|\eZ\Publish\Core\MVC\Symfony\Matcher\ConfigurableMatcherFactoryInterface */
```

Comment:
Nitpick: type hints here are inconsistent with the most of ezpublish-kernel.

Reply:
We discussed this and I think we agreed to follow one coding standard in all of repositories

Reply:
Since I'm the only one who dislikes and can't read those, I can agree _as long as someone opens PR for kernel making it consistent_. Right now we have mix, with multi-line being majority.